### PR TITLE
Bringing in ImmutableJS 3.8.2 

### PR DESCRIPTION
**Pull request by vickramdhawal**
Bringing in ImmutableJS 3.8.2 which is release under MIT license. Source of immutableJS https://github.com/facebook/immutable-js/releases/tag/v3.8.2.
Also updating the gitignore file to exclude node_modules inside src/JSUtils

### CODE REVIEW

**Comment by nethip** [comment]
Thanks for the PR @vickramdhawal :+1: 

Did you get a chance to run unit tests?

**Comment by nethip** [comment]
Also looks like CLA has not been signed that is why the travis tests are failing. Can you signup using this link? Please sign the CLA here http://dev.brackets.io/brackets-contributor-license-agreement.html.

**Comment by vickramdhawal** [comment]
I've signed up now. I did run the unit tests, but have to bear with multiple dialogs to get it to completion. Will update once its done.

**Comment by nethip** [comment]
Thanks @vickramdhawal I have restarted the build.

**Comment by nethip** [comment]
You can disable the dialog popups by returning from the below function. We will have to figure out other means of handling the dialog popups.
https://github.com/adobe/brackets/blob/master/src/document/DocumentCommandHandlers.js#L1727

**Comment by vickramdhawal** [comment]
immutableJS is used only by the FileTreeView & the FileTreeViewModel and the tests are passing for both. Unit tests are also passing except for the ones already failing on master.

**Comment by swmitra** [comment]
Good work @vickramdhawal.

**Review by swmitra — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,8 @@ package-lock.json
 
 # ignore node_modules inside src
 /src/node_modules
+/src/JSUtils/node_modules
+/src/JSUtils/node/node_modules
```
Comment (ficristo) [inline .gitignore]: IMO this fixes the effects and not the cause of the problem.

**Pull request merged**